modules/dicts/rehashing_map.py

```diff
@@ -0,0 +1,50 @@
+from .dictionary import Dictionary
+
+#using a hash function
+class RehashMapDictionary(Dictionary):
+
+
+    def __init__(self, num_bins):   
+        self.num_bins = num_bins
+        self.bin = []
+        for i in range(self.num_bins):
+            self.bin.append([]) 
+        self.size = 0 #total number of elements
+
+    def add(self,key,value):
+        index = hash(key) %  self.num_bins     
+        
+        for i, (k,v) in enumerate(self.bin[index]):
+            if k == key:
+                self.bin[index][i] = (key,value)  #updates the existing key
+                return
+        else:
```

Comment:
oh very interesting use of for-else. You actually don't need it since you have a return statement in your for loop though!